// @neynar/nodejs-sdk/build/neynar-api/v2/openapi-farcaster/apis/subscribers-api.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Farcaster API V2
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 2.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscribersApi = exports.SubscribersApiFactory = exports.SubscribersApiFp = exports.SubscribersApiAxiosParamCreator = void 0;
const axios_1 = __importDefault(require("axios"));
// Some imports not used depending on template conditions
// @ts-ignore
const common_1 = require("../common");
// @ts-ignore
const base_1 = require("../base");
/**
 * SubscribersApi - axios parameter creator
 * @export
 */
const SubscribersApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Fetch what fids and contracts a fid is subscribed to.
         * @summary Fetch what a given fid is subscribed to
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProvider} subscriptionProvider
         * @param {number} [viewerFid]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        subscribedTo: async (apiKey, fid, subscriptionProvider, viewerFid, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('subscribedTo', 'apiKey', apiKey);
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('subscribedTo', 'fid', fid);
            // verify required parameter 'subscriptionProvider' is not null or undefined
            (0, common_1.assertParamExists)('subscribedTo', 'subscriptionProvider', subscriptionProvider);
            const localVarPath = `/farcaster/user/subscribed_to`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (viewerFid !== undefined) {
                localVarQueryParameter['viewer_fid'] = viewerFid;
            }
            if (subscriptionProvider !== undefined) {
                localVarQueryParameter['subscription_provider'] = subscriptionProvider;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetch subscribers for a given fid\'s contracts. Doesn\'t return addresses that don\'t have an fid.
         * @summary Fetch subscribers for a given fid
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProviders} subscriptionProvider
         * @param {number} [viewerFid]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        subscribers: async (apiKey, fid, subscriptionProvider, viewerFid, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('subscribers', 'apiKey', apiKey);
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('subscribers', 'fid', fid);
            // verify required parameter 'subscriptionProvider' is not null or undefined
            (0, common_1.assertParamExists)('subscribers', 'subscriptionProvider', subscriptionProvider);
            const localVarPath = `/farcaster/user/subscribers`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (viewerFid !== undefined) {
                localVarQueryParameter['viewer_fid'] = viewerFid;
            }
            if (subscriptionProvider !== undefined) {
                localVarQueryParameter['subscription_provider'] = subscriptionProvider;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
        /**
         * Fetch created subscriptions for a given fid\'s.
         * @summary Fetch created subscriptions for a given fid
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProvider} subscriptionProvider
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        subscriptionsCreated: async (apiKey, fid, subscriptionProvider, options = {}) => {
            // verify required parameter 'apiKey' is not null or undefined
            (0, common_1.assertParamExists)('subscriptionsCreated', 'apiKey', apiKey);
            // verify required parameter 'fid' is not null or undefined
            (0, common_1.assertParamExists)('subscriptionsCreated', 'fid', fid);
            // verify required parameter 'subscriptionProvider' is not null or undefined
            (0, common_1.assertParamExists)('subscriptionsCreated', 'subscriptionProvider', subscriptionProvider);
            const localVarPath = `/farcaster/user/subscriptions_created`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, common_1.DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            if (fid !== undefined) {
                localVarQueryParameter['fid'] = fid;
            }
            if (subscriptionProvider !== undefined) {
                localVarQueryParameter['subscription_provider'] = subscriptionProvider;
            }
            if (apiKey != null) {
                localVarHeaderParameter['api_key'] = String(apiKey);
            }
            (0, common_1.setSearchParams)(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: (0, common_1.toPathString)(localVarUrlObj),
                options: localVarRequestOptions,
            };
        },
    };
};
exports.SubscribersApiAxiosParamCreator = SubscribersApiAxiosParamCreator;
/**
 * SubscribersApi - functional programming interface
 * @export
 */
const SubscribersApiFp = function (configuration) {
    const localVarAxiosParamCreator = (0, exports.SubscribersApiAxiosParamCreator)(configuration);
    return {
        /**
         * Fetch what fids and contracts a fid is subscribed to.
         * @summary Fetch what a given fid is subscribed to
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProvider} subscriptionProvider
         * @param {number} [viewerFid]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async subscribedTo(apiKey, fid, subscriptionProvider, viewerFid, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.subscribedTo(apiKey, fid, subscriptionProvider, viewerFid, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetch subscribers for a given fid\'s contracts. Doesn\'t return addresses that don\'t have an fid.
         * @summary Fetch subscribers for a given fid
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProviders} subscriptionProvider
         * @param {number} [viewerFid]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async subscribers(apiKey, fid, subscriptionProvider, viewerFid, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.subscribers(apiKey, fid, subscriptionProvider, viewerFid, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
        /**
         * Fetch created subscriptions for a given fid\'s.
         * @summary Fetch created subscriptions for a given fid
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProvider} subscriptionProvider
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        async subscriptionsCreated(apiKey, fid, subscriptionProvider, options) {
            const localVarAxiosArgs = await localVarAxiosParamCreator.subscriptionsCreated(apiKey, fid, subscriptionProvider, options);
            return (0, common_1.createRequestFunction)(localVarAxiosArgs, axios_1.default, base_1.BASE_PATH, configuration);
        },
    };
};
exports.SubscribersApiFp = SubscribersApiFp;
/**
 * SubscribersApi - factory interface
 * @export
 */
const SubscribersApiFactory = function (configuration, basePath, axios) {
    const localVarFp = (0, exports.SubscribersApiFp)(configuration);
    return {
        /**
         * Fetch what fids and contracts a fid is subscribed to.
         * @summary Fetch what a given fid is subscribed to
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProvider} subscriptionProvider
         * @param {number} [viewerFid]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        subscribedTo(apiKey, fid, subscriptionProvider, viewerFid, options) {
            return localVarFp.subscribedTo(apiKey, fid, subscriptionProvider, viewerFid, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetch subscribers for a given fid\'s contracts. Doesn\'t return addresses that don\'t have an fid.
         * @summary Fetch subscribers for a given fid
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProviders} subscriptionProvider
         * @param {number} [viewerFid]
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        subscribers(apiKey, fid, subscriptionProvider, viewerFid, options) {
            return localVarFp.subscribers(apiKey, fid, subscriptionProvider, viewerFid, options).then((request) => request(axios, basePath));
        },
        /**
         * Fetch created subscriptions for a given fid\'s.
         * @summary Fetch created subscriptions for a given fid
         * @param {string} apiKey API key required for authentication.
         * @param {number} fid
         * @param {SubscriptionProvider} subscriptionProvider
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        subscriptionsCreated(apiKey, fid, subscriptionProvider, options) {
            return localVarFp.subscriptionsCreated(apiKey, fid, subscriptionProvider, options).then((request) => request(axios, basePath));
        },
    };
};
exports.SubscribersApiFactory = SubscribersApiFactory;
/**
 * SubscribersApi - object-oriented interface
 * @export
 * @class SubscribersApi
 * @extends {BaseAPI}
 */
class SubscribersApi extends base_1.BaseAPI {
    /**
     * Fetch what fids and contracts a fid is subscribed to.
     * @summary Fetch what a given fid is subscribed to
     * @param {string} apiKey API key required for authentication.
     * @param {number} fid
     * @param {SubscriptionProvider} subscriptionProvider
     * @param {number} [viewerFid]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SubscribersApi
     */
    subscribedTo(apiKey, fid, subscriptionProvider, viewerFid, options) {
        return (0, exports.SubscribersApiFp)(this.configuration).subscribedTo(apiKey, fid, subscriptionProvider, viewerFid, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetch subscribers for a given fid\'s contracts. Doesn\'t return addresses that don\'t have an fid.
     * @summary Fetch subscribers for a given fid
     * @param {string} apiKey API key required for authentication.
     * @param {number} fid
     * @param {SubscriptionProviders} subscriptionProvider
     * @param {number} [viewerFid]
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SubscribersApi
     */
    subscribers(apiKey, fid, subscriptionProvider, viewerFid, options) {
        return (0, exports.SubscribersApiFp)(this.configuration).subscribers(apiKey, fid, subscriptionProvider, viewerFid, options).then((request) => request(this.axios, this.basePath));
    }
    /**
     * Fetch created subscriptions for a given fid\'s.
     * @summary Fetch created subscriptions for a given fid
     * @param {string} apiKey API key required for authentication.
     * @param {number} fid
     * @param {SubscriptionProvider} subscriptionProvider
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof SubscribersApi
     */
    subscriptionsCreated(apiKey, fid, subscriptionProvider, options) {
        return (0, exports.SubscribersApiFp)(this.configuration).subscriptionsCreated(apiKey, fid, subscriptionProvider, options).then((request) => request(this.axios, this.basePath));
    }
}
exports.SubscribersApi = SubscribersApi;
